dropshot/build.rs

```diff
@@ -13,15 +13,21 @@
 // limitations under the License.
 //
 
+// NOTE: The `usdt` crate uses inline assembly, which prior to Rust 1.59 or 1.66
+// on macOS, required the use of nightly features. These crate-local features
+// are used to determine which toolchain is used to compile Dropshot, and then
+// include the correct `#![feature(...)]` directives on the basis of that.
 fn main() {
     println!("cargo:rerun-if-changed=build.rs");
 
+    // `feature(asm)` is stable.
     if version_check::is_min_version("1.59").unwrap_or(false) {
         println!("cargo:rustc-cfg=usdt_stable_asm");
     }
 
-    // Once asm_sym is stablilized, add an additional check so that those
-    // building on macos can use the stable toolchain with any hassle.
-    //
-    // A matching rust-cfg option named `usdt_stable_asm_sym` seems appropriate.
+    // `feature(asm_sym)` is stable, which is required on macOS.
```

Comment:
this is fine, but FWIW we do it a little differently in usdt:

https://github.com/oxidecomputer/usdt/blob/master/probe-test-build/build.rs#L23-L28

Reply:
Hmm, good point. I _think_ this is all logically the same, just inverting everything, right?